Sources/SwiftCurrent_SwiftUI/Models/WorkflowViewModel.swift

```diff
@@ -8,37 +8,43 @@
 
 import SwiftCurrent
 import SwiftUI
+import Combine
 
 @available(iOS 14.0, macOS 11, tvOS 14.0, watchOS 7.0, *)
 final class WorkflowViewModel: ObservableObject {
-    @Published var body = AnyView(EmptyView())
+    @Published var body: Any?
+    var onAbandonPublisher = PassthroughSubject<Void, Never>()
     var isLaunched: Binding<Bool>?
-    var onAbandon = [() -> Void]()
+
+    init(isLaunched: Binding<Bool>) {
+        self.isLaunched = isLaunched
+    }
 }
 
 @available(iOS 14.0, macOS 11, tvOS 14.0, watchOS 7.0, *)
 extension WorkflowViewModel: OrchestrationResponder {
     func launch(to destination: AnyWorkflow.Element) {
-        extractView(from: destination).model = self
+        body = extractView(from: destination).erasedView
     }
 
     func proceed(to destination: AnyWorkflow.Element, from source: AnyWorkflow.Element) {
-        extractView(from: destination).model = self
+        body = extractView(from: destination).erasedView
     }
 
     func backUp(from source: AnyWorkflow.Element, to destination: AnyWorkflow.Element) {
-        extractView(from: destination).model = self
+        body = extractView(from: destination).erasedView
     }
 
     func abandon(_ workflow: AnyWorkflow, onFinish: (() -> Void)?) {
         isLaunched?.wrappedValue = false
-        onAbandon.forEach { $0() }
+        body = nil
```

Comment:
I'm not sure I understand this part.  Why `nil`?

Reply:
Because the `Workflow` is now gone, so any remnants of the views should be gone as well.

Reply:
I'll leave this one unresolved in case that answer is somehow unsatisfactory...Suspect this would be pretty easy to understand with animation cruft.

Reply:
So a `nil` view is different than an `EmptyView`?

Reply:
Absolutely `nil` is the absence of a value. With our recursive crazy this essentially means nothing is provided as a View to display, allowing for maximum SwiftUI jiggery pokery optimization from Apple. `EmptyView` is a concrete type meaning "I don't have a value". I suspect, but do not know for certain, that it probably itself is just an `if false` statement.

Reply:
👍  I wonder what animation weirdness this is going to cause, but I wonder that for everything SwiftUI.